[Shashank Acharya] iP

sheshenk opened the pull request:
# Lurch - Your Inarticulate Assistant

> "You rang?" - Lurch

Lurch is a CLI assistant with the demeanour of the butler from The Addams Family. Lurch is

- text-based
- easy to use
- ~fast~ super-fast to use!

...and it's **FREE**.

## Installation

1. Download the latest release from [here](https://github.com/sheshenk/ip/releases)
2. Double click the `.jar` file.
3. Add your tasks with commands like `todo`, `deadline`, `event`.
4. Let Lurch do the rest. 😉

## Features

- [X] Manage tasks
- [ ] Reminders (Coming Soon)

## Customising Lurch

Lurch's source files are located [here](https://github.com/sheshenk/ip).

The main entry point for Lurch is in the `Duke.java` file.

```java
public static void main(String[] args) {
    new Duke(saveFilePath).run();
}
```

Commit:
Add Gradle support

sheshenk committed:
Level-1

sheshenk committed:
Level-2

sheshenk committed:
Added Task class

sheshenk committed:
Added marking & unmarking increments

sheshenk committed:
Level-4: Inheritance

sheshenk committed:
A-TextUiTesting

sheshenk committed:
Level-5: Added error handling

sheshenk committed:
Level-6: Deletion

sheshenk committed:
Update .gitignore

sheshenk committed:
Level-7: Saving functions

sheshenk committed:
Level-8: LocalDate

sheshenk committed:
Merge pull request #1 from sheshenk/branch-Level-7

Merge saving branch

sheshenk committed:
Merge pull request #2 from sheshenk/branch-Level-8

Merge datetime branch

sheshenk committed:
Update files to include more OOP

sheshenk committed:
Update code to use packages

sheshenk committed:
Merge pull request #3 from nus-cs2103-AY2223S1/add-gradle-support

Add Gradle support

sheshenk committed:
Merge branch 'master' of https://github.com/sheshenk/ip

sheshenk committed:
Add JUnit tests

sheshenk committed:
Update build.gradle

sheshenk committed:
Update meta for JAR packaging

sheshenk committed:
Add JavaDocs

sheshenk committed:
Update to coding standard

sheshenk committed:
Update JavaDocs for TaskList

sheshenk committed:
Add find feature

sheshenk committed:
Merge pull request #4 from sheshenk/branch-A-JavaDoc

Branch a java doc

sheshenk committed:
Merge pull request #5 from sheshenk/branch-A-CodingStandard

Update to coding standard

sheshenk committed:
Merge branch 'master' into branch-Level-9

sheshenk committed:
Merge pull request #6 from sheshenk/branch-Level-9

Add find feature

sheshenk committed:
Add checkstyle

sheshenk committed:
Add GUI (without command support)

sheshenk committed:
Add basic UI implementation with Lurch commands

sheshenk committed:
Remove CLI implementation

sheshenk committed:
Add FXML formatting

sherylkong18 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,109 @@
+public class Duke {
+
+    // MESSAGES
+    private static final String markMessage = "Nice! I've marked this task as done:";
```
display messages can be handled in UI class

sherylkong18 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,109 @@
+                String direction = currentCommand.getDirection();
+                String meta = currentCommand.getMeta();
+                switch (direction) {
+                    case "bye":
```
parsing user inputs can be handled under parser class for more concise code

sherylkong18 commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,120 @@
+import duke.DukeException;
+import java.util.ArrayList;
+
+public class TaskList {
```
Can add a brief description of what a TaskList does

sherylkong18 commented on src/main/java/duke/task/Todo.java:
```diff
@@ -0,0 +1,17 @@
+package duke.task;
+
+public class Todo extends Task {
```
Can add a brief description about what Todo does

sherylkong18 reviewed (commented):
Overall looks good,
code can be organised better by passing the data to the different classes to handle them (eg. let UI class handle user input, let Parser class handle parsing any texts)

sheshenk committed:
Merge pull request #7 from sheshenk/branch-A-CheckStyle

Add checkstyle

sheshenk committed:
Move all classes into packages

sheshenk committed:
Add Message enum

sheshenk committed:
Create command package

sheshenk committed:
Modify Parser to return Command object

sheshenk committed:
Remove tests

sheshenk committed:
Update Ui and Message

sheshenk committed:
Remove parsing functions from Task classes

sheshenk committed:
Update storage format

sheshenk committed:
Add date parsing format

sheshenk committed:
Update bot messages

sheshenk committed:
Update FXML files

sheshenk committed:
Add javadocs

sheshenk committed:
Merge pull request #8 from sheshenk/branch-Level-10

Branch level 10

sheshenk committed:
Update checkstyle

sheshenk committed:
Update build.gradle

sheshenk committed:
Add assertions

These assertions ensure non-primitives data types like
* String
* Command
* Task
are not `null` when they are used.

sheshenk committed:
Remove obvious comments

sheshenk committed:
Merge pull request #9 from sheshenk/branch-A-CodeQuality

Remove obvious comments

sheshenk committed:
Merge pull request #10 from sheshenk/branch-A-Assertions

Add assertions

sheshenk committed:
Create .yml file for continuous integration

sheshenk committed:
Merge pull request #11 from sheshenk/branch-A-CI

Create .yml file for continuous integration

sheshenk committed:
Add getters and setters for time in Task

sheshenk committed:
Add cases to parse snooze command

sheshenk committed:
Add messages for snooze command

sheshenk committed:
Add snooze command

To give users of Lurch more flexibility over their tasks, this
commit introduces functionality to let them snooze their tasks.

To use the new command, simply enter `snooze /for` followed by
an integer value denoting the amount of time, and a character
(either `M`, `D`, `h`, `s`) to denote the degree of time.

> `30D` represents 30 days, `25m` represents 25 minutes, etc.

sheshenk committed:
Add checkstyle fixes

sheshenk committed:
Merge pull request #12 from sheshenk/branch-B-Snooze

Branch b snooze

sheshenk committed:
Improve GUI

Used Scene Builder to modify anchor points and add colors.

sheshenk committed:
Change messages for personality

Made the bot seem more like Lurch.

sheshenk committed:
Handle empty input

Add a new error message to handle empty input

sheshenk committed:
Merge pull request #14 from sheshenk/branch-A-BetterGui

Improve GUI

sheshenk committed:
Merge pull request #15 from sheshenk/branch-A-Personality

Change messages for personality

sheshenk committed:
Merge pull request #16 from sheshenk/branch-A-MoreErrorHandling

Handle empty input

sheshenk committed:
Add title

sheshenk committed:
Update docs

sheshenk committed:
Update for release
